registry/handlers/app.go

```diff
@@ -458,11 +466,20 @@ func (app *App) authorized(w http.ResponseWriter, r *http.Request, context *Cont
 	if err != nil {
 		switch err := err.(type) {
 		case auth.Challenge:
+			// NOTE(duglin):
+			// Since err.ServeHTTP will set the HTTP status code for us
+			// we need to set the content-type here.  The serveJSON
+			// func will try to do it but it'll be too late at that point.
+			// I would have have preferred to just have the auth.Challenge
+			// ServerHTTP func just add the WWW-Authenticate header and let
+			// serveJSON set the HTTP status code and content-type but I wasn't
+			// sure if that's an ok design change. STEVVOOE ?
```

Comment:
That design change is acceptable, but let's do in a separate PR. We'd want to change the auth package to *only* set the header and not the response code.